docs/reference/src/main/asciidoc/configure.asciidoc

```diff
@@ -273,6 +273,37 @@ The functionality is implemented as a built-in portable extension processing all
 |`org.jboss.weld.bootstrap.vetoTypesWithoutBeanDefiningAnnotation` || A regular expression. If a non-empty string, then all annotated types whose `AnnotatedType#getJavaClass().getName()` matches this pattern are vetoed if not annotated with a bean defining annotation.
 |=======================================================================
 
+[[remove-unused-beans]]
+==== Memory consumption optimization - removing unused beans
+
+CDI applications consist of user-defined beans implementing the business logic, but also beans coming from libraries (e.g. DeltaSpike) and integrations (e.g. various Java EE technologies - JSF, Bean Validation, Batch).
+Most applications actually use only a small part of the beans provided by libraries and integrations.
+Still, Weld has to retain the metadata for all beans in memory!
+What a waste.
+If <<allow-optimized-cleanup>> is allowed Weld can now remove the _unused_ beans after bootstrap.
+
+An _unused_ bean:
+
+* is not excluded by the configuration properties described below,
+* is not a built-in bean, session bean, extension, interceptor or decorator,
+* does not have a name
+* does not declare an observer
+* is not eligible for injection to any injection point,
+* does not declare a producer which is eligible for injection to any injection point,
+* is not eligible for injection into any `Instance` injection point.
+
+NOTE: As usual, there is a trade-off between memory consumption and bootstrap time. The tests showed that for a rather small application deployed to WildFly the memory consumption was reduced by approx. 10% (300 KB) at the cost of increasing the bootstrap time by approx. 5% (60ms).
```

Comment:
Ok, we could use something more general, e.g.: _"The results varies depending on the application"_.

Reply:
That's a good sentence, or we could incorporate the fancy phrase `Your mileage may vary` :)
Maybe start with your sentence and follow up with `Using Weld Numberguess sample<link> with added DS libraries, we have measured....` so as to provide some specific data but still underline that it ain't guaranteed.

Reply:
I'd like to avoid mentioning numberguess and DS.

Reply:
As long as we want to mention exact numbers, we should relate that to exact deployment, shouldn't we? Otherwise it feels like numbers conjured out of blue.

Reply:
So maybe we can just say that there is a trade-off and that you will likely gain less memory consumption but increase boot time and that's that. The rest can go to some blog post?

Reply:
As I said, let's stick with some more general sentence. From what I know about perf testing it's probably wise to keep all results in secret ;-)